packages/extensions/README.md

```diff
@@ -0,0 +1,180 @@
+
+## Prerequisites
+
+Before starting, ensure you have the following installed:
+
+- [Docker](https://www.docker.com/get-started)
+- [Node.js](https://nodejs.org/)
+- [Git](https://git-scm.com/)
+
+## Getting Started
+
+You can either create a new repository or fork this repository to build your DiracX extension. Follow one of the methods below:
+
+### Method 1: Fork the Repository
+
+1. **Fork this repository** on GitHub.
+2. **Clone the forked repository** to your local machine:
+
+   ```bash
+   git clone https://github.com/YOUR_USERNAME/diracx-web.git
+   cd diracx-web/packages/extensions
+   npm install
+   ```
+
+### Method 2: Create a New Next.js Project
+
+1. **Create a new Next.js project** using the following command:
+
+   ```bash
+   npx create-next-app your-extension
+   cd your-extension
+   ```
+
+2. **Add the DiracX Web Components Library** to your project:
+
+   ```bash
+   npm install @dirac-grid/diracx-web-components
+   ```
+
+   To include additional dependencies in your project, such as `cypress` for end-to-end tests, you can follow these steps:
+
+   Run the following command to install the desired dependency:
+
+   ```bash
+   npm install <dependency-name>
+   ```
+
+   Add the `--save-dev` or `-D` flag to install the dependency as a development dependency.
+
+   Replace `<dependency-name>` with the name of the dependency you want to install.
+
+   For example, to install `cypress`, run:
+
+   ```bash
+   npm install cypress -D
+   ```
+
+   Make sure to consult the documentation of each dependency for further instructions on how to use them in your project.
+
+   For `cypress`, you can refer to the [Cypress documentation](https://docs.cypress.io/guides/overview/why-cypress.html) for detailed usage instructions.
+
+3. **Add the OIDC postinstall script** to your `package.json` file to copy necessary service worker files:
+
+   ```json
+   "scripts": {
+     "postinstall": "node ./node_modules/@axa-fr/react-oidc/bin/copy-service-worker-files.mjs public"
+   }
+   ```
+
+   This is needed to update the service worker files when the OIDC library version changes.
+
+   See the [OIDC library documentation](https://github.com/AxaFrance/oidc-client/tree/main/packages/react-oidc#getting-started) for more information.
+
+4. **Modify the app pages** to use components from the `diracx-components` library (e.g., providers, apps).
+
+### Architecture
+
+We strongly recommend following the directory structure below to keep your project organized:
+
+- `/src/app`: Contains the main application logic and Next.js setup. This directory houses the core of the application built using Next.js, where each page.tsx file represents a page in the application with [folder-based routing](https://nextjs.org/docs/app/building-your-application/routing). [Next.js Official Documentation](https://nextjs.org/docs)
```

Comment:
I think you can even provide more details about `src/app`: the subsections such as `dashboard` and `auth` (`layout` and `page`), their role and what is required to have a working extension.
This could also be (entirely/partially) explained in `Method2`.